pcl_ros/CMakeLists.txt

```diff
@@ -1,245 +1,89 @@
-  cfg/Feature.cfg
-  cfg/Filter.cfg
-  cfg/MLS.cfg
-  cfg/SACSegmentation.cfg
-  cfg/SACSegmentationFromNormals.cfg
-  cfg/SegmentDifferences.cfg
-  cfg/StatisticalOutlierRemoval.cfg
-  cfg/RadiusOutlierRemoval.cfg
-  cfg/VoxelGrid.cfg
-  cfg/CropBox.cfg
-)
-
-## Declare the catkin package
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES
-    pcl_ros_filters
-    pcl_ros_io
-    pcl_ros_tf
-  CATKIN_DEPENDS
-    dynamic_reconfigure
-    message_filters
-    nodelet
-    nodelet_topic_tools
-    pcl_conversions
-    pcl_msgs
-    rosbag
-    roscpp
-    sensor_msgs
-    std_msgs
-    tf
-  DEPENDS
-    Boost
-    EIGEN3
-    PCL
+  tf2
+  tf2_geometry_msgs
+  tf2_ros
+  EIGEN3
+  PCL
 )
 
 ## Declare the pcl_ros_tf library
 add_library(pcl_ros_tf src/transforms.cpp)
-target_link_libraries(pcl_ros_tf ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-add_dependencies(pcl_ros_tf ${catkin_EXPORTED_TARGETS})
-
-## Nodelets
-
-## Declare the pcl_ros_io library
-add_library(pcl_ros_io
-  src/pcl_ros/io/bag_io.cpp
-  src/pcl_ros/io/concatenate_data.cpp
-  src/pcl_ros/io/concatenate_fields.cpp
-  src/pcl_ros/io/io.cpp
-  src/pcl_ros/io/pcd_io.cpp
-)
-target_link_libraries(pcl_ros_io pcl_ros_tf ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-class_loader_hide_library_symbols(pcl_ros_io)
-
-## Declare the pcl_ros_features library
-add_library(pcl_ros_features
-  src/pcl_ros/features/feature.cpp
-  # Compilation is much faster if we include all the following CPP files in feature.cpp
-  src/pcl_ros/features/boundary.cpp
-  src/pcl_ros/features/fpfh.cpp
-  src/pcl_ros/features/fpfh_omp.cpp
-  src/pcl_ros/features/shot.cpp
-  src/pcl_ros/features/shot_omp.cpp
-  src/pcl_ros/features/moment_invariants.cpp
-  src/pcl_ros/features/normal_3d.cpp
-  src/pcl_ros/features/normal_3d_omp.cpp
-  src/pcl_ros/features/pfh.cpp
-  src/pcl_ros/features/principal_curvatures.cpp
-  src/pcl_ros/features/vfh.cpp
+ament_target_dependencies(pcl_ros_tf
+  "pcl_conversions"
+  "rclcpp"
+  "sensor_msgs"
+  "tf2"
+  "tf2_geometry_msgs"
+  "tf2_ros"
 )
-target_link_libraries (pcl_ros_features ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-add_dependencies(pcl_ros_features ${PROJECT_NAME}_gencfg)
-class_loader_hide_library_symbols(pcl_ros_features)
-
-
-## Declare the pcl_ros_filters library
-add_library(pcl_ros_filters
-  src/pcl_ros/filters/extract_indices.cpp
-  src/pcl_ros/filters/filter.cpp
-  src/pcl_ros/filters/passthrough.cpp
-  src/pcl_ros/filters/project_inliers.cpp
-  src/pcl_ros/filters/radius_outlier_removal.cpp
-  src/pcl_ros/filters/statistical_outlier_removal.cpp
-  src/pcl_ros/filters/voxel_grid.cpp
-  src/pcl_ros/filters/crop_box.cpp
-)
-target_link_libraries(pcl_ros_filters pcl_ros_tf ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-add_dependencies(pcl_ros_filters ${PROJECT_NAME}_gencfg)
-class_loader_hide_library_symbols(pcl_ros_filters)
-
-## Declare the pcl_ros_segmentation library
-add_library (pcl_ros_segmentation
```

Comment:
Please don't delete this block, just comment it out. Its good to have a base to start from for others to port